### DISCUSSION TITLE
IDP listing not working

### DISCUSSION THREAD

**Discussion by pcMarcos** [sventorben/keycloak-home-idp-discovery#390] [Q&A]
hey, 

I need some help and would appreciate if you could help me. 

I have multiple IDP configuration with the same domain, but I'm facing some issues to list the IDPs. the option Forward to first matched IdP is disabled, as well as other configurations, but it is still redirecting to the first matched. 

```
{"alias":"IDP discovery","config":{"userAttribute":"email","bypassLoginPage":false,"forwardToLinkedIdp":false,"forwardToFirstMatch":"false"}}
```

![image](https://github.com/sventorben/keycloak-home-idp-discovery/assets/33794434/401e6561-d947-4de2-9dfd-98f70adc965b)

**Comment by sventorben** [comment]
Which version of the extension and Keycloak are you using? Would you mind exporting your realm config and share it?

**Comment by pcMarcos** [comment]
I am using the version `21.3.0` of the extension and the `21.1.2` version of Keycloak. 

I discovered something interesting. The IDP was not being listed when I configured the Authenticatin flow but didn't save the IDP discovery configurations. But after saving it with an alias, it started working.

Here my screenshot. The left is the export of my realm when it was not working and the right was when I opened the IDP discovery configuration, set an alias and saved. 

![image (5)](https://github.com/sventorben/keycloak-home-idp-discovery/assets/33794434/a692af64-06cb-47cf-9386-80e48b7118b5)
